src/tracers/blob_saver_tracer.rs

```diff
@@ -0,0 +1,102 @@
+use super::tracer::Tracer;
+use crate::{eravm_error::EraVmError, state::VMState, value::FatPointer, Opcode};
+use std::collections::HashMap;
+use u256::{H160, H256, U256};
+use zkevm_opcode_defs::ethereum_types::Address;
+
+#[derive(Default, Debug)]
+pub struct BlobSaverTracer {
+    pub blobs: HashMap<U256, Vec<U256>>,
+}
+
+impl BlobSaverTracer {
+    pub fn new() -> Self {
+        Self {
+            blobs: Default::default(),
+        }
+    }
+}
+
+const CONTRACT_DEPLOYER_ADDRESS: Address = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x80, 0x06,
+]);
+
+const KNOWN_CODES_STORAGE_ADDRESS: Address = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x80, 0x04,
+]);
+
+pub(crate) fn hash_evm_bytecode(bytecode: &[u8]) -> H256 {
+    use zkevm_opcode_defs::sha2::{Digest, Sha256};
```

Comment:
Please put imports at the top of the file.

Reply:
Done